### PR TITLE
refactor: move uber/mock dependency to submodules

### PR DESCRIPTION
**Pull request by erdemtuna**
Hello, this change addresses #453.

### CODE REVIEW

File: mock/go.mod [hunk 1]
```diff
@@ -0,0 +1,12 @@
+module mock
```
Comment (rueian) [inline mock/go.mod:1]:
```suggestion
module github.com/redis/rueidis/mock
```

File: rueidishook/go.mod [hunk 2]
```diff
@@ -0,0 +1,12 @@
+module rueidishook
```
Comment (rueian) [inline rueidishook/go.mod:1]:
```suggestion
module github.com/redis/rueidis/rueidishook
```

**Review by rueian — COMMENTED** [review]
Hi @erdemtuna,

Many thanks! Just that those module names should be fully specified.

**Comment by erdemtuna** [comment]
missed those, thank you @rueian

**Comment by rueian** [comment]
LGTM. Thanks again!

**Pull request merged**